Yoast/yoast-components
# Toggle - cornerstone toggle improvements

## Summary

This PR can be summarized in the following changelog entry:

* Toggle component accessibility and RTL support improvements

## Relevant technical choices:

- adds a switch to RTL to all the demo pages in the standalone app
- updates the `dir` attribute on the `<html>` element when switching to RTL, this partially addresses #665
- adds Toggle and CornerstoneToggle demos to the "UI Controls" tab (formerly "Checkbox" tab) in the demo pages
- Toggle: removes the `<label>` element and uses `aria-labelledby` instead
- Toggle: passes the ID as a prop with a default value
- CornerstoneToggle: changes the text to "Mark as cornerstone content" (addresses design feedback)
- Toggle: uses `getRtlStyle()` for RTL languages and adjusts various CSS 
- Toggle: prevents the page from scrolling when using the Spacebar 
- Toggle: when using the Spacebar, switch the toggle on `keyup` instead of `keydown`, as that's the native behavior
- Toggle: makes `onToggleDisabled()` work when using the Spacebar 
- Toggle: uses `aria-disabled` when the Toggle is disabled 

## Test instructions

This PR can be tested by following these steps:

- go in the standalone app (yarn start go to localhost:3333)
- go in the new "UI Controls" tab
- verify the Toggle and CornerstoneToggle demos work as expected
- switch to RTL pressing the button at the top of the page
- verify again the Toggle and CornerstoneToggle demos work as expected

Also, please verify the style adjustments from #682 are OK.

Fixes #681
